engine/src/main/java/io/camunda/zeebe/streamprocessor/ProcessingStateMachine.java

```diff
@@ -259,17 +264,44 @@ private void processCommand(final LoggedEvent command) {
     final var processingStartTime = ActorClock.currentTimeMillis();
     processingTimer = metrics.startProcessingDurationTimer(metadata.getRecordType());
 
+    final AtomicReference<ProcessingResult> processingResultRef = new AtomicReference<>();
```

Comment:
:wrench: If we only use it to extract the result from the lambda we pass to `zeebeDbTransaction.run`, then I don't think it needs to be atomic. AFAIK the callback runs inline without switching threads, so you could use something like Agrona's `MutableReference` (they also have `MutableInteger`, `MutableLong`, etc., for cases where you don't need atomicity).

That said, a more interesting alternative might be that we have something like `ZeebeTransaction#call` which would return some result. A little out of scope, especially if we don't have many such usages, but something to keep in mind.